### PR TITLE
Select Tool: Deselect selected layer when clicked while holding shift

### PR DESCRIPTION
**Pull request by akozlev**
Fixes "In the Select tool, Shift-clicking an already selected layer does not remove it from the selection. (As posted in Discord)" 
From #989

### CODE REVIEW

File: editor/src/messages/tool/tool_messages/select_tool.rs [hunk 1]
```diff
@@ -430,6 +430,17 @@ impl Fsm for SelectToolFsmState {
 
 						RotatingBounds
 					} else if intersection.last().map(|last| selected.contains(last)).unwrap_or(false) {
+						if input.keyboard.get(add_to_selection as usize) {
+							let last_selected = intersection.last().unwrap();
```
Comment (Keavon) [inline editor/src/messages/tool/tool_messages/select_tool.rs:434]: @TrueDoctor is this `unwrap` safe? Also if you can review the rest of this small PR, that'd be appreciated. Thanks.

**Comment by Keavon** [comment]
A small UX improvement request for this: I want to be able to Shift-drag a layer without it getting deselected (when dragging, the Shift modifier is used to constrain the drag to the X or Y axis). So instead of your feature working by deselecting upon the mousedown, it should ideally work, instead, on the mouseup as long as the user didn't perform a mousemove following the mousedown. Thanks.

**Comment by Keavon** [comment]
I'm converting this back to a draft for the moment (pending the request in the above comment) so I can more easily keep track of which PRs are awaiting my attention for merging right now. Please just ping me when you're ready again for a review with that change, thank you!!

**Review by TrueDoctor — APPROVED** [review]
LGTM

**Review by Keavon — APPROVED** [review]
Works great!

**Pull request merged**